pkg/machines/components/vm/overview/vmOverviewCard.jsx

```diff
@@ -214,13 +220,24 @@ class VmOverviewCard extends React.Component {
             </DescriptionListDescription>
         );
 
+        let cpuEditButton = (
+            <Button variant="link" className="edit-inline" isInline isAriaDisabled={!vm.persistent || !this.state.virtXMLAvailable} onClick={this.openCpuType}>
+                {_("edit")}
+            </Button>
+        );
+        if (!this.state.virtXMLAvailable) {
+            cpuEditButton = (
+                <Tooltip id='virt-install-missing'
+                         content={_("virt-install package needs to be installed on the system in order to edit this attribute")}>
+                    {cpuEditButton}
+                </Tooltip>
```

Comment:
I like how much simpler this approach is:)
However, I want to stop for a second and think about this `virt-xml` tool. Is it pulled as dependency by something? Do we expect it to be present in majority of cases? Should we depend on it on our own?

Reply:
virt-xml is part of virt-install packagel. no extra dependency is needed. virt-install is already a recommeneded dependenncy https://github.com/cockpit-project/cockpit/blob/master/tools/cockpit.spec#L655

Users can already use cockpit-machines without having virt-install package installed, but these will cause them to not be able to create new VMs. Now they won't be able to edi CPU model either, they would see the tooltip prompting them to install virt-install. 

I think it's a fair trade off.
Pro:
----- 
One one side virt-xml will simplyfy rastically our code removing all XML editing functions
 (especially once I port all XML editing to this tool)

I expect it to be more error phrone, it's a tool shipped by the virt-team which will replace my/Simon's code which can always contain bugs. 

virt-xml is much more sophisticated than our dummy XML creation, there is a lot of tweaks that can be done for increasing VM performance which I never implement because I am not familiar

Cotra:
-------- 
 Some users which decide to not install 'virt-install', they will just be able to perform lifecycle operations on existing XMs but not really create new ones or edit the configuration.

Reply:
Cool, if it is part of virt-install and also it is recommended by us, then this is surely fine :) The amount of installations where users would opt out is going to be minimal and since you show nice warning about this, I think this is completely fine and good approach!